### ISSUE TITLE
EfficientDet D0 weight SIZE is 40MB???

### ISSUE THREAD

**Issue by jduki** [toandaominh1997/EfficientDet.Pytorch#100]
As I read paper, efficient d0 size should be 3.8MB.
Why d0 checkpoint weight  included on git has 40MB size ??

**Comment by midasklr** [comment]
> As I read paper, efficient d0 size should be 3.8MB.
> Why d0 checkpoint weight included on git has 40MB size ??

I guess 3.8M, which is 3.9M in the paper, is the number of parameters(see table 2...#Params),not the extact store space in the disk.

**Comment by nsarang** [comment]
> > As I read paper, efficient d0 size should be 3.8MB.
> > Why d0 checkpoint weight included on git has 40MB size ??
> 
> I guess 3.8M, which is 3.9M in the paper, is the number of parameters(see table 2...#Params),not the extact store space in the disk.

Then, shouldn't it be around 16MB since the weights are in float32?

**Comment by midasklr** [comment]
> > > As I read paper, efficient d0 size should be 3.8MB.
> > > Why d0 checkpoint weight included on git has 40MB size ??
> > 
> > 
> > I guess 3.8M, which is 3.9M in the paper, is the number of parameters(see table 2...#Params),not the extact store space in the disk.
> 
> Then, shouldn't it be around 16MB since the weights are in float32?

yes,u r right. the weight file is a orderdict include both weights and bias and other param names (orderdict keys)and their values(around 3.9m ).

**Comment by dvlshah** [comment]
It is 40M because it stores the optimizer state during training in the weight file itself. Store only state dict to disk and you will get the exact weight.

**Comment by guanshuicheng** [comment]
> As I read paper, efficient d0 size should be 3.8MB.
> Why d0 checkpoint weight included on git has 40MB size ??

3.8M明显是参数的数量
